### PR TITLE
Better output types for upscale nodes

### PR DESCRIPTION
**Pull request by RunDevelopment**
This adds an alias type that correctly calculates the number of output channels of an upscaled image.

### CODE REVIEW

File: src/common/types/typedef.ts [hunk 1]
```diff
@@ -272,6 +280,56 @@ const addBuiltinTypes = (definitions: TypeDefinitions) => {
...
+     *   outputChannels: int(1..inf),
+     * } = match imageChannels {
+     *   1 => 1,
+     *   4 => match inputChannels { 3 => add(outputChannels, 1), _ => outputChannels },
+     *   _ => outputChannels
+     * }
+     * ```
+     */
+    definitions.add(
+        new AliasDefinition(
```
Comment (RunDevelopment) [inline src/common/types/typedef.ts:297]: I'm honestly considering adding a parser for type expressions and statements. I mean, look at this!

Comment (joeyballentine) [reply src/common/types/typedef.ts:297]: That's probably a good idea. The one you did for text append was pretty gnarly as well

Comment (RunDevelopment) [reply src/common/types/typedef.ts:297]: Yeah, I think I'll throw some ANTLR4 stuff together. Or do you know any other parser generators for JS/TS?

Comment (joeyballentine) [reply src/common/types/typedef.ts:297]: I do not.

**Review by joeyballentine — APPROVED** [review]

**Pull request merged**